tests/end2end/features/downloads.feature

```diff
@@ -522,6 +522,17 @@ Feature: Downloading things from a website.
         And I open data/downloads/download2.bin without waiting
         Then the download prompt should be shown with "(tmpdir)/downloads/subdir/download2.bin"
 
+    Scenario: Clearing the last download directory when changing download location
+        When I set downloads.location.prompt to true
+        And I set downloads.location.suggestion to both
+        And I set downloads.location.remember to true
+        And I open data/downloads/download.bin without waiting
+        And I wait for the download prompt for "*/download.bin"
+        And I run :prompt-accept (tmpdir)(dirsep)downloads(dirsep)subdir
+        And I set downloads.location.directory to (tmpdir)
```

Comment:
`(tmpdir)` isn't expanded here:
```
Traceback (most recent call last):
  File "/home/jay/Code/qutebrowser/qutebrowser/config/configcommands.py", line 55, in _handle_config_error
    yield
  File "/home/jay/Code/qutebrowser/qutebrowser/config/configcommands.py", line 135, in set
    save_yaml=not temp)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/config.py", line 461, in set_str
    converted = opt.typ.from_str(value)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/configtypes.py", line 251, in from_str
    self.to_py(value)  # for validation
  File "/home/jay/Code/qutebrowser/qutebrowser/config/configtypes.py", line 1452, in to_py
    value, "must be a valid directory!")
qutebrowser.config.configexc.ValidationError: Invalid value '(tmpdir)' - must be a valid directory!

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jay/Code/qutebrowser/qutebrowser/commands/runners.py", line 329, in run_safely
    self.run(text, count)
  File "/home/jay/Code/qutebrowser/qutebrowser/commands/runners.py", line 308, in run
    result.cmd.run(self._win_id, args, count=count)
  File "/home/jay/Code/qutebrowser/qutebrowser/commands/command.py", line 551, in run
    self.handler(*posargs, **kwargs)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/configcommands.py", line 135, in set
    save_yaml=not temp)
  File "/usr/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/configcommands.py", line 57, in _handle_config_error
    raise cmdutils.CommandError(str(e))
qutebrowser.api.cmdutils.CommandError: Invalid value '(tmpdir)' - must be a valid directory!
19:06:20.657 ERROR    message    message:error:58 Invalid value '(tmpdir)' - must be a valid directory!
```

You should be able to fix it by creating a one-off command (although I'm not sure if that's the best way), or by making this a unit test. See `test_downloads_bdd.py` for the command definitions here (searching for `(tmpdir)` in that file is helpful)

Reply:
I made this a one-off set command instead to fix the tmpdir expansion